app.rb

```diff
@@ -0,0 +1,155 @@
+require './book'
+require './classroom'
+require './base_decorator'
+require './person'
+require './rental'
+require './student'
+require './teacher'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def start
+    selecting
+  end
+
+  def books_list
+    puts 'No books available!!' if @books.empty?
+    puts "There are #{@books.count} books available"
+
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+    end
+  end
+
+  def peoples_list
+    puts 'No added people!! Please Add new person.' if @people.empty?
+    puts "#{@books.count} peoples added."
+
+    @people.each_with_index do |person, index|
+      puts "#{index + 1})[#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number:'
+    choice = gets.chomp
+    print 'Name:'
+    name = gets.chomp
+    print 'Age:'
+    age = gets.chomp.to_i
+    case choice
+    when '1'
+      print 'Class:'
+      classroom = gets.chomp
+      create_student(age, classroom, name)
+    when '2'
+      print 'Specialization:'
+      specialization = gets.chomp
+      create_teacher(age, name, specialization)
+    else
+      puts 'Invalid, please try again'
+    end
+  end
+
+  def create_student(age, classroom, name)
+    print 'Parent permission? [Y/N]'
+    option = gets.chomp.downcase
+    case option
+    when 'y'
+      student = Student.new(age, classroom, name, parent_permission: true)
+    when 'n'
+      student = Student.new(age, classroom, name, parent_permission: false)
+    else
+      puts 'Invalid option, please try again'
+      return
+    end
+    @people << student
+    puts ''
+    puts "Student is created successfully.Student ID is #{student.id}"
+  end
```

Comment:
- [ ] The app crashes and quits when I try to create a student. Kindly fix that.

  ![levy-2](https://user-images.githubusercontent.com/81468675/174326670-6ecadc3a-d0b4-4080-9d63-62932bfc3588.jpg)